runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -141,8 +156,10 @@ public void invoke() {
                     throw new IllegalStateException("Orchestrator can not be selected");
                 }
 
-            } catch (NetworkException | TimeoutException | IllegalStateException e) {
-                log.warn("WorkflowRequest: Error while running {} on attempt {}, cause:", this, x, e);
+            } catch (TimeoutException e) {
+                log.warn("WorkflowRequest: Error while running {} on attempt {}, cause:", this, retryIteration, e);
+            } catch (RuntimeException e) {
+                handleException(e);
```

Comment:
For your consideration - I personally vote for getting rid of handleException, it reduces the complexity of the code if you make it flat

Reply:
Let me make a note of your suggestion @xnull